README.md

```diff
@@ -7,17 +7,163 @@
   <img src="http://i.imgur.com/COd11eL.png" />
 </div>
 
+`ember-parachute` is an addon that improves upon the experience of working with
+query params in Ember. Instead of defining query params in both your route and controller, with this addon you can define them in one place as a query param map/object.
+
+This map is the source of truth for your query params, and will generate a mixin that you can then add into your controller. The mixin adds very helpful properties and methods that makes working with query params a breeze!
+
+One important point about this addon is that it is opinionated about _where_ the data is fetched. In a traditional query params setup, your route is responsible for fetching data in its `model` hook. With this addon, the responsibility moves into the controller. The benefit of this approach is that data fetching no longer blocks your UI from loading, and paves the way for advanced UX such as ["skeleton UI"][skeleton-ui].
+
 ## Installation
 
 ```
 ember install ember-parachute
 ```
 
+## Usage
+
+The source of truth for your application's query params are query param maps. First, define one:
+
+```js
+// query-params/my-query-params.js
+import QueryParams from 'ember-parachute';
+
+export const myQueryParams = {
+  parachuteOpen: {
+    as: 'parachute',
+    defaultValue: true,
+  },
+  page: {
+    defaultValue: 1,
+    refresh: true
+  },
+  search: {
+    defaultValue: '',
+    refresh: true
+  },
+  tags: {
+    defaultValue: [ 'Ember', 'Parachute' ]
+  }
+}
+
+export default new QueryParams(myQueryParams);
+```
+
+This builds a new `QueryParams` instance that you can now use in your controller. In this example we're using `ember-concurrency`, but it is *not* required:
+
+```js
+// controllers/my-route.js
+import Ember from 'ember';
+import { task } from 'ember-concurrency';
+import myQueryParams from '../query-params/my-query-params';
+
+const {
+  computed: { or },
+  inject: { service },
+  Controller
+} = Ember;
+
+export default Controller.extend(myQueryParams.Mixin, {
+  qp: service(),
+
+  queryParamsChanged: or('queryParamsState.{page,search,tags}.changed'),
+
+  queryParamsDidChange({ shouldRefresh, queryParams }) {
+    // if any query param with `refresh: true` is changed, `shouldRefresh` is `true`
+    if (shouldRefresh) {
+      this.get('fetchModel').perform();
+    }
+  },
+
+  fetchModel: task(function *() {
+    yield // fetch something with query params
+  }).restartable(),
+
+  actions: {
+    resetAll() {
+      // reset all query params to their default values specified in the query param map
+      this.get('qp').resetParamsFor(this);
```

Comment:
The `qp` service only accepts the `routeName` string. The `QueryParams` **class** static methods accept only the controller. 

I think this may be a bit confusing. We might need to circle back on the API